content/en/blog/_posts/2021-04-21-Graceful-Node-Shutdown-Beta.md

```diff
@@ -0,0 +1,66 @@
+kubelet-node ~ # systemd-inhibit --list
+    Who: kubelet (UID 0/root, PID 1515/kubelet)
+    What: shutdown
+    Why: Kubelet needs time to handle node shutdown
+    Mode: delay
+
+1 inhibitors listed.
+```
+
+One important consideration we took when designing this feature is that not all pods are created equal. For example, some of the pods running on a node such as a logging related daemonset should stay running as long as possible to capture important logs during the shutdown itself. As a result, pods are split into two categories: “regular” and “critical”. [Critical pods](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/#marking-pod-as-critical) are those that have `priorityClassName` set to `system-cluster-critical` or `system-node-critical`; all other pods are considered regular. In our example, the logging DaemonSet would run as a critical pod. During the graceful shutdown, regular pods are terminated first, followed by critical pods. As an example, this would allow a critical pod like those part of a logging daemonset to continue functioning, and collecting logs during the termination of regular pods.
+
+## How do I use it?
+Graceful node shutdown is controlled with the `GracefulNodeShutdown` [feature gate](https://github.com/kubernetes/website/blob/master/docs/reference/command-line-tools-reference/feature-gates) which is enabled by default in 1.21.
+
+Graceful node shutdown feature is configured with two kubelet configuration options: `ShutdownGracePeriod`, `ShutdownGracePeriodCriticalPods`. These can be adjusted by editing the kubelet config file that is passed to kubelet via the `--config` flag; for more details, refer to [set kubelet parameters via a configuration file](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/).
+
+During a shutdown, kubelet terminates pods in two phases. You can configure how long each of these phases lasts.
+1. Terminate regular pods running on the node.
+2. Terminate critical pods running on the node.
+
+The settings that control the duration of shutdown are:
+* `ShutdownGracePeriod`
+    * Specifies the total duration that the node should delay the shutdown by. This is the total grace period for pod termination for both regular and critical pods.
+* `ShutdownGracePeriodCriticalPods`
+    * Specifies the duration used to terminate critical pods during a node shutdown. This should be less than   ShutdownGracePeriod.
+
+For example, if `ShutdownGracePeriod=30s`, and `ShutdownGracePeriodCriticalPods=10s`, kubelet will delay the node shutdown by 30 seconds. During the shutdown, the first 20 (30-10) seconds would be reserved for gracefully terminating normal pods, and the last 10 seconds would be reserved for terminating critical pods.
+
+## How can I learn more?
+* Documentation: [https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown](https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown)
```

Comment:
```suggestion

* Read the [documentation](/docs/concepts/architecture/nodes/#graceful-node-shutdown) for graceful node shutdown)
￼* Read the enhancement proposal, [KEP 2000](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown)
￼* View the [code](https://github.com/kubernetes/kubernetes/tree/release-1.21/pkg/kubelet/nodeshutdown) on GitHub
```

Reply:
Done (agree, the active action reads better here) :)